src/main/java/infrastructure/SocketConnection.java

```diff
@@ -19,16 +19,15 @@ public String read() throws IOException {
 
     @Override
     public void write(String output) throws IOException {
-        String formattedOutput = output + "\n";
-        socket.getOutputStream().write(formattedOutput.getBytes());
+        socket.getOutputStream().write(output.getBytes());
     }
 
     @Override
     public void close() throws IOException {
-        this.socket.close();
+        this.socket.shutdownOutput();
```

Comment:
What is the motivation here? Does this mean we don't ever close the socket completely?

Reply:
I believe the cachedthreadpoolexecutor closes it after being idle for 60 sec

Reply:
Here's the excerpt: In general, applications implementing graceful closes will first close their output channels and then wait for the peer on the other side of the connection to close its output channels. When both sides are done telling each other they won’t be sending any more data (i.e., closing output channels), the connection can be closed fully, with no risk of reset.

Reply:
Another excerpt:  If the input channel isn’t closed by the peer within some timeout period, the application may force connection close to save resources

Reply:
I think that's where the cachedthreadpoolexecutor comes into play

Reply:
Here's why closing the input channel is risky:  Closing the input channel of your connection is riskier, unless you know the other side doesn’t plan to send any more data. If the other side sends data to your closed input channel, the operating system will issue a TCP “connection reset by peer” message

Reply:
Gotcha! Thanks for all of that. The thread pool executor will clean up the thread, so I guess the connection will get cleaned up when garbage collection happens? It might be good to verify that if you haven't.